browser/playlist/android/playlist_page_handler_factory.cc

```diff
@@ -0,0 +1,54 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/playlist/android/playlist_page_handler_factory.h"
+
+#include <utility>
+
+#include "brave/browser/brave_browser_process.h"
+#include "brave/browser/playlist/playlist_service_factory.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/profiles/incognito_helpers.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/keyed_service/content/browser_context_dependency_manager.h"
+
+namespace playlist {
+
+// static
+PlaylistPageHandlerFactory* PlaylistPageHandlerFactory::GetInstance() {
+  return base::Singleton<PlaylistPageHandlerFactory>::get();
+}
+
+// static
+mojo::PendingRemote<mojom::PageHandler>
+PlaylistPageHandlerFactory::GetForContext(content::BrowserContext* context) {
+  return static_cast<PlaylistAndroidPageHandler*>(
+             GetInstance()->GetServiceForBrowserContext(context, true))
+      ->MakeRemote();
+}
+
+PlaylistPageHandlerFactory::PlaylistPageHandlerFactory()
+    : BrowserContextKeyedServiceFactory(
+          "PlaylistAndroidPageHandler",
+          BrowserContextDependencyManager::GetInstance()) {
+  DependsOn(playlist::PlaylistServiceFactory::GetInstance());
+}
+
+PlaylistPageHandlerFactory::~PlaylistPageHandlerFactory() = default;
+
+KeyedService* PlaylistPageHandlerFactory::BuildServiceInstanceFor(
+    content::BrowserContext* context) const {
+  auto* profile = Profile::FromBrowserContext(context);
+  playlist::PlaylistServiceFactory::GetInstance()->GetForBrowserContext(
+      context);
+  return new PlaylistAndroidPageHandler(profile);
+}
+
+content::BrowserContext* PlaylistPageHandlerFactory::GetBrowserContextToUse(
+    content::BrowserContext* context) const {
+  return chrome::GetBrowserContextRedirectedInIncognito(context);
+}
```

Comment:
I think we shouldn't use original profile from incognito window. It'll mix up url loader, which is not recommened by our security guideline.

Reply:
I realize that we're not creating PlaylistService on incognito(OTR) profile. And I guess this was what security folks suggested. https://github.com/brave/security/issues/971#issuecomment-1212348522